src/scanning/scanner.ts

```diff
@@ -15,10 +15,15 @@
 import {Analyzer} from '../analyzer';
 import {ScannedFeature} from '../model/model';
 import {ParsedDocument} from '../parser/document';
+import {Warning} from '../warning/warning';
+
+export interface ScanResult {
```

Comment:
my reaction to reading this: are these all warnings found during scanning? Or are these just feature-unrelated warnings? If there's a warning on a feature, does it show up in the warnings array, in the feature.warnings property, or both?

Perhaps rename `warnings` to `otherWarnings` or something that clarifies?